### ISSUE TITLE
Support for "UserAssigned" type in identity block in azurerm_logic_app_standard

### ISSUE THREAD

**Issue by mg-8** [hashicorp/terraform-provider-azurerm#18057]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Logic App Standard supports  **"UserAssigned"** , in addition, to SystemAssigned identity type if deployed through Azure portal. However, the current AzureRM provider **3.19.1** does not support **"UserAssigned"**. 

Please include this feature at your earliest so we can avoid using **"azapi_resource**" block to create logic app standard with **"UserAssigned"** identity which is cumbersome to use and manage. 

### New or Affected Resource(s)/Data Source(s)

azurerm_logic_app_standard

### Potential Terraform Configuration

```hcl
resource "azurerm_logic_app_standard" "logic-app" {

  name                = var.name
  resource_group_name = var.rg_name
  location            = var.rg_location
  app_service_plan_id            = azurerm_service_plan.app-svc-plan.id
  storage_account_name       = azurerm_storage_account.storage-account.name
  storage_account_access_key = sensitive(azurerm_storage_account.storage-account.primary_access_key)
   
  identity {
      type         = "UserAssigned"
      identity_ids = var.identity_ids
  }

  site_config {} 

}


│ Error: expected identity.0.type to be one of [SystemAssigned], got UserAssigned
│
│   with module.logic_app.azurerm_logic_app_standard.logic-app,
│   on ..\modules\logic-app\main.tf line 100, in resource "azurerm_logic_app_standard" "logic-app":
│  100:       type         = "UserAssigned"
```

**Comment by mikegouldthorp** [comment]
I submitted a similar issue.

https://github.com/hashicorp/terraform-provider-azurerm/issues/16715

**Comment by ziyeqf** [comment]
this has been added by PR #18109

**Comment by tombuildsstuff** [comment]
Fixed via #18109
